src/tools/web/elektrad/router.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/gorilla/mux"
+
+	elektra "go.libelektra.org/kdb"
+)
+
+func setupRouter() http.Handler {
+	r := mux.NewRouter()
+	r.Use(handleMiddleware)
+
+	r.HandleFunc("/version", getVersionHandler).Methods("GET")
+
+	r.HandleFunc("/kdbFind/{path:.*}", getFindHandler).Methods("GET")
+
+	r.HandleFunc("/kdbMv/{path:.*}", postMoveHandler).Methods("POST")
+
+	r.HandleFunc("/kdbMeta/{path:.*}", postMetaHandler).Methods("POST")
+	r.HandleFunc("/kdbMeta/{path:.*}", deleteMetaHandler).Methods("DELETE")
+
+	r.HandleFunc("/kdb", getKdbHandler).Methods("GET")
+	r.HandleFunc("/kdb/{path:.*}", getKdbHandler).Methods("GET")
+	r.HandleFunc("/kdb/{path:.*}", putKdbHandler).Methods("PUT")
+	r.HandleFunc("/kdb/{path:.*}", deleteKdbHandler).Methods("DELETE")
+
+	return r
+}
+
+func parseKeyNameFromURL(r *http.Request) string {
+	vars := mux.Vars(r)
+	keyName := vars["path"]
+
+	if len(keyName) == 0 {
+		return "/"
+	}
+
+	return keyName
+}
+
+func stringBody(r *http.Request) (string, error) {
+	value := ""
+
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&value); err != nil {
+		return "", err
+	}
+
+	return value, nil
+}
+
+func getKeySet(handle elektra.KDB, key elektra.Key) (elektra.KeySet, error) {
```

Comment:
No docu, what is this function supposed to do?

And is router.go the correct file?